adapters/yssp/yssp.go

```diff
@@ -0,0 +1,213 @@
+package yssp
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type YSSPAdapter struct {
+	URI string
+}
+
+func (a *YSSPAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errors := make([]error, 0, 1)
+
+	if len(request.Imp) == 0 {
+		err := &errortypes.BadInput{
+			Message: "No impression in the bid request",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	reqs := make([]*adapters.RequestData, 0, len(request.Imp))
+	headers := http.Header{}
+
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	if request.Device != nil && request.Device.UA != "" {
+		headers.Set("User-Agent", request.Device.UA)
+	}
+
+	for idx, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		err := json.Unmarshal(imp.Ext, &bidderExt)
+		if err != nil {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: ext.bidder not provided", idx),
+			}
+			errors = append(errors, err)
+			continue
+		}
+
+		var ysspExt openrtb_ext.ExtImpYSSP
+		err = json.Unmarshal(bidderExt.Bidder, &ysspExt)
+		if err != nil {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: %s", idx, err.Error()),
+			}
+			errors = append(errors, err)
+			continue
+		}
+
+		if ysspExt.Dcn == "" {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: missing param dcn", idx),
+			}
+			errors = append(errors, err)
+			continue
+		}
+
+		if ysspExt.Pos == "" {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: missing param pos", idx),
+			}
+			errors = append(errors, err)
+			continue
+		}
```

Comment:
If you add a minimum length in the json schema file, you can remove the Dcn and Pos empty string checks.